src/main/java/liquibase/ext/clickhouse/sqlgenerator/CreateDatabaseChangeLogTableClickHouse.java

```diff
@@ -39,30 +43,56 @@ public boolean supports(CreateDatabaseChangeLogTableStatement statement, Databas
     return database instanceof ClickHouseDatabase;
   }
 
+  private String generateSqlTemplate(boolean replicated) {
+    StringBuilder query = new StringBuilder();
+    query.append("CREATE TABLE IF NOT EXISTS %s.%s ");
+
+    if (replicated) query.append("ON CLUSTER '%s' ");
+
+    query.append("(");
+    query.append("ID String,");
+    query.append("AUTHOR String,");
+    query.append("FILENAME String,");
+    query.append("DATEEXECUTED DateTime64,");
+    query.append("ORDEREXECUTED UInt64,");
+    query.append("EXECTYPE String,");
+    query.append("MD5SUM Nullable(String),");
+    query.append("DESCRIPTION Nullable(String),");
+    query.append("COMMENTS Nullable(String),");
+    query.append("TAG Nullable(String),");
+    query.append("LIQUIBASE Nullable(String),");
+    query.append("CONTEXTS Nullable(String),");
+    query.append("LABELS Nullable(String),");
+    query.append("DEPLOYMENT_ID Nullable(String)) ");
+
+    if (replicated) query.append("ENGINE ReplicatedMergeTree('%s','%s') ORDER BY ID");
+    else query.append("ENGINE MergeTree() ORDER BY ID");
+
+    return query.toString();
+  }
+
   @Override
   public Sql[] generateSql(
       CreateDatabaseChangeLogTableStatement statement,
       Database database,
       SqlGeneratorChain sqlGeneratorChain) {
+    Map<String, String> properties = ParamsLoader.getLiquibaseClickhouseProperties();
+    boolean isReplicated = properties != null;
```

Comment:
same here, it is not directly understandable why properties != null => isReplicated